app/2.0/docs/devguide/custom-elements.md

```diff
@@ -316,29 +375,84 @@ MyMixin = function(superClass) {
+```js
+MyMixin = (superClass) => class extends superClass {
+  ...
+}
+```
 
-Add a mixin to your element like this:
+The mixin class can define properties, observers, and methods just like a regular element class. In
+addition, a mixin can incorporate other mixins:
 
 ```js
-class MyElement extends MyMixin(Polymer.Element) {
-  static get is() { return 'my-element' }
+MyCompositeMixin = (base) => class extends MyMixin2(MyMixin1(base)) {
+  ...
 }
 ```
 
-This creates a new class defined by the `MyMixin` factory, so the inheritance hierarchy is:
+Because mixins are simply adding classes to the inheritance chain, all of the usual rules of
+inheritance apply. For example, mixin classes can define constructors, can call superclass methods
+with `super`, and so on.
 
-```
-MyElement <= MyMixin(Polymer.Element) <= Polymer.Element
+**Document your mixins.** The Polymer build and lint tools require some extra documentation tags
+to property analyze mixins and elements that use them. Without the documentation tags, the tools
+will log warnings. For details on documenting mixins, see [Class mixins](../tools/documentation#class-mixins)
+in Document your elements.
+{.alert .alert-info}
+
+
+### Packaging mixins for sharing
+
+When creating a mixin that you intend to share with other groups or publish, a couple of additional
+steps are recommended:
+
+-   Use the [`Polymer.dedupingMixin`](/{{{polymer_version_dir}}}/docs/api/#function-Polymer.dedupingMixin)
+    function to produce a mixin that can only be applied once.
+
+-   Create a unique namespace for your mixins, to avoid colliding with other mixins or classes that
+    might have similar names.
+
+The `dedupingMixin` function is useful because a mixin that's used by other mixins may accidentally
+be applied more than once. For example if `MixinA` includes `MixinB` and `MixinC`, and you create an element
+that uses `MixinA` but also uses `MixinB` directly:
+
+```js
+class MyElement extends MixinB(MixinA(Polymer.Element)) { ... }
 ```
 
-You can apply multiple mixins in sequence:
+At this point, your element contains two copies of `MixinB` in its  prototype chain. `dedupingMixin`
+takes a mixin function as an argument, and returns a new, deduplicating mixin function:
 
 ```js
-class AnotherElement extends AnotherMixin(MyMixin(Polymer.Element)) { … }
+dedupingMixinB = Polymer.dedupingMixin(mixinB);
 ```
 
+The deduping mixin has two advantages: first, whenever you use the mixin, it memoizes the generated
+class, so any subsequent uses on the same base class return the same class object—a minor optimization.
+
+More importantly, the deduping mixin checks whether this mixin has already been applied anywhere in
+the base class's prototype chain. If it has, the mixin simply returns the base class. In the example
+above, if you used `dedupingMixinB` instead of  `mixinB` in both places, the mixin would only be
+applied once.
+
+The following example shows one way you might create a namespaced, deduping mixin,
```

Comment:
>The following example shows one way you might create a namespaced, deduping mixin,

The following example shows one way you might create a namespaced, deduping mixin: